llvm/lib/Support/RWMutex.cpp

```diff
@@ -107,6 +108,14 @@ RWMutexImpl::unlock()
   return errorcode == 0;
 }
 
+bool RWMutexImpl::try_lock() {
+  pthread_rwlock_t *rwlock = static_cast<pthread_rwlock_t *>(data_);
+  assert(rwlock != nullptr);
+
+  int errorcode = pthread_rwlock_tryrdlock(rwlock);
```

Comment:
There's an inconsistency here. `RWMutexImpl` has `shared` variants of `lock` and `unlock` that are used for reading while the plain versions of `lock` and `unlock` are for writing. I think this needs to use `pthread_rwlock_trywrlock` and then you probably want to add `RWMutexImpl::try_lock_shared()`.